util/libcrypto.num

```diff
@@ -4660,139 +4660,170 @@ OSSL_PROVIDER_unload                    4615	3_0_0	EXIST::FUNCTION:
 OSSL_PROVIDER_add_builtin               4616	3_0_0	EXIST::FUNCTION:
 OSSL_PROVIDER_get_param_types           4617	3_0_0	EXIST::FUNCTION:
 OSSL_PROVIDER_get_params                4618	3_0_0	EXIST::FUNCTION:
-d2i_OSSL_CRMF_ENCRYPTEDVALUE            4619	3_0_0	EXIST::FUNCTION:CMP
-i2d_OSSL_CRMF_ENCRYPTEDVALUE            4620	3_0_0	EXIST::FUNCTION:CMP
-OSSL_CRMF_ENCRYPTEDVALUE_free           4621	3_0_0	EXIST::FUNCTION:CMP
```

Comment:
Something seems to have gone wrong here. We shouldn't be deleting already existing entries in this file.

Reply:
Due to renaming exported functions and adding new exported functions, which sometimes happens even during a PR due to requested changes, we sometimes need to adapt `util/libcrypto.num`. Note that we adapt lines in the file, but we make sure not add lines indicating that a function is gone (NOEXIST).
I typically do this by manually calling `make ordinals`, or is there a better way?
Nevertheless I suppose intermediate changes to the file should be harmless since we can squash them later, right?

BTW, keeping this file consistent is quite a hassle IMHO, in particular when there are concurrent additions by other contributors. Moreover, `util/mknum.pl` appears to be somewhat buggy - it sometimes needs to be be called twice in order to get stable output on the auto-declared ASN.1 `*_it` symbols.

Reply:
I've just re-built `util/libcrypto.num` and squashed all our recent commits that made changes to this file, such that now the changes to the file are in the style you expected: simple additions at its end.

Reply:
I just found that I had to repeat the step described above because of the mentioned bug in `util/mknum.pl`: I needs to be called twice in order to get a stable output (which gets compared by Travis CI). The changes done by the second invocation are in our case:
```
diff --git a/util/libcrypto.num b/util/libcrypto.num
index d6c3b86204..7c624966af 100644
--- a/util/libcrypto.num
+++ b/util/libcrypto.num
@@ -4796,17 +4796,20 @@ EVP_MD_fetch                            4743    3_0_0   EXIST::FUNCTION:
 EVP_set_default_properties              4744   3_0_0   EXIST::FUNCTION:
 OSSL_PARAM_construct_end                4745   3_0_0   EXIST::FUNCTION:
 EC_GROUP_check_named_curve              4746   3_0_0   EXIST::FUNCTION:EC
-OSSL_CMP_PKISTATUS_it                   4747   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_PKISTATUS_it                   4747   3_0_0   NOEXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_PKISTATUS_it                   4748   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
 OSSL_CMP_PKISTATUS_it                   4748   3_0_0   EXIST:EXPORT_VAR_AS_FUNCTION:FUNCTION:CMP
 d2i_OSSL_CMP_PKIHEADER                  4749   3_0_0   EXIST::FUNCTION:CMP
 i2d_OSSL_CMP_PKIHEADER                  4750   3_0_0   EXIST::FUNCTION:CMP
 OSSL_CMP_PKIHEADER_free                 4751   3_0_0   EXIST::FUNCTION:CMP
 OSSL_CMP_PKIHEADER_new                  4752   3_0_0   EXIST::FUNCTION:CMP
-OSSL_CMP_PKIHEADER_it                   4753   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_PKIHEADER_it                   4753   3_0_0   NOEXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_PKIHEADER_it                   4754   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
 OSSL_CMP_PKIHEADER_it                   4754   3_0_0   EXIST:EXPORT_VAR_AS_FUNCTION:FUNCTION:CMP
 d2i_OSSL_CMP_MSG                        4755   3_0_0   EXIST::FUNCTION:CMP
 i2d_OSSL_CMP_MSG                        4756   3_0_0   EXIST::FUNCTION:CMP
-OSSL_CMP_MSG_it                         4757   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_MSG_it                         4757   3_0_0   NOEXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
+OSSL_CMP_MSG_it                         4758   3_0_0   EXIST:!EXPORT_VAR_AS_FUNCTION:VARIABLE:CMP
 OSSL_CMP_MSG_it                         4758   3_0_0   EXIST:EXPORT_VAR_AS_FUNCTION:FUNCTION:CMP
 OSSL_CMP_ITAV_set0                      4759   3_0_0   EXIST::FUNCTION:CMP
 OSSL_CMP_ITAV_get0_type                 4760   3_0_0   EXIST::FUNCTION:CMP
```